File: LeetCode/intersection_of_two_arrays_ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> hash;
        vector<int> res;
        
        for(int n : nums2)
            hash[n]++;
        
        for(int n : nums1) {
            if(hash[n] > 0) {
                res.push_back(n);
                hash[n]--;
            }
        }
        
        return res;
    }
};

File: LeetCode/intersection_of_two_arrays_ii.go
func intersect(nums1 []int, nums2 []int) []int {
    freq := map[int]int{}

    for _, num := range nums1 {
        freq[num]++
    }

    ans := []int{}
    for _, num := range nums2 {
        if freq[num] > 0 {
            ans = append(ans, num)
            freq[num]--
        }
    }

    return ans
}
